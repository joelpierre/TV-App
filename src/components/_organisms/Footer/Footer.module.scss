.Footer {
  width: 100%;
  background-color: $color-tint-psi;
  padding-top: spacing(4);
  padding-bottom: spacing(4);
  position: relative;
  z-index: 100;

  &__container {
    @include max-width-container;
  }

  &__content {
    text-align: left;
  }

  &__copy {
    font-size: 14px;
    color: $color-tint-alpha;
  }

  &__link {
    color: $color-tint-alpha;
    text-decoration: underline;

    @include hover-focus {
      color: $color-tint-alpha;
      text-decoration: none;
    }
  }
}
