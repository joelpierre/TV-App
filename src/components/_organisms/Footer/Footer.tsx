import classNames from 'classnames';
import React from 'react';

import styles from './Footer.module.scss';

interface IFooterProps {
  className?: string;
}

const Footer: React.FunctionComponent<IFooterProps> = ({
  className
}) => {
  return (
    <footer className={classNames(styles.Footer, className)}>
      <div className={styles.Footer__container}>
        <p className={styles.Footer__copy}>
          Created by Joel Pierre, {new Date().getFullYear()} |
          <a className={styles.Footer__link}
            href="https://github.com/joelpierre">https://github.com/joelpierre</a>
        </p>
      </div>
    </footer>
  );
};

export default Footer;
