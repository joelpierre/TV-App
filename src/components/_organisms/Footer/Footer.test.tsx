import React from 'react';
import { render, RenderResult } from 'test-utils';

import Footer from './Footer';

const setup = (props = {}): RenderResult => {
  const setupProps = { ...props };
  return render(<Footer {...setupProps} />);
};

describe('<Footer/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
