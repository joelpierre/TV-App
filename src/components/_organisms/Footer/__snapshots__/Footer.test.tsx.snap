// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="Footer"
  >
    <div
      class="Footer__container"
    >
      <p
        class="Footer__copy"
      >
        Created by Joel Pierre, 2020 |
        <a
          class="Footer__link"
          href="https://github.com/joelpierre"
        >
          https://github.com/joelpierre
        </a>
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
